### PR TITLE
WIP: async derive literal

### PR DESCRIPTION
**Pull request by tanhauhau**
This MR is to make `deriveLiteral` async, so can support asynchronous plugins and more!

TODO List
* [x] fix test

@Cryrivers need your help to set up the test integrations 😭

### CODE REVIEW

**Comment by Cryrivers** [comment]
This PR is awesome! Let me try fixing failed test cases

发自我的 iPhone

File: packages/@manta-style/runtime/webpack.config.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   mode: 'production',
   entry: __dirname + '/src/index.ts',
   devtool: 'source-map',
+  target: 'node',
```
[the remark this answers is not in the document]
Comment (tanhauhau) [reply packages/@manta-style/runtime/webpack.config.js]: this is temp patch for making the node api available for the plugin system

**Comment by Cryrivers** [comment]
Okay, let me figure out the package organization today. I feel both `runtime` and `plugins` should be able to run on Node.js and Browser. So they are need to be built in `umd` instead of `commonjs` and not rely on Node APIs directly.

**Comment by tanhauhau** [comment]
@Cryrivers yea i understand. i think the discovery should be taken out from the plugin-system itself

File: packages/@manta-style/runtime/webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tanhauhau) [reply packages/@manta-style/runtime/webpack.config.js]: @Cryrivers i've fixed this.

**Pull request merged**